src/main/java/io/supertokens/cronjobs/Cronjobs.java

```diff
@@ -91,9 +91,9 @@ public static void addCronjob(Main main, CronTask task) {
         }
         Cronjobs instance = getInstance(main);
         synchronized (instance.lock) {
-            instance.executor.scheduleWithFixedDelay(task, task.getInitialWaitTimeSeconds(),
-                    task.getIntervalTimeSeconds(), TimeUnit.SECONDS);
             if (!instance.tasks.contains(task)) {
+                instance.executor.scheduleWithFixedDelay(task, task.getInitialWaitTimeSeconds(),
+                        task.getIntervalTimeSeconds(), TimeUnit.SECONDS);
                 instance.tasks.add(task);
```

Comment:
add test for this

Reply:
And why does the addCronjob get called so many times for the same job anyway?

Reply:
Also, it's possible that the task obj is a different instance of the same class?